### PR TITLE
Adapt Polygon proposal to Mumbai

### PR DESCRIPTION
**Pull request by kevincheng96**
This adapts most of the initial Polygon proposal to be run on Mumbai. Namely, we'll be able to test updating the configuration of Mumbai Comet, bridging over USDC reserves, and updating the ENS text record.

Two actions are left out of the Mumbai proposal:

1. Bridging COMP: This is because Polygon recently changed their token mapping process from using `RootChainManager` to `FxRoot`, so we'll be unable to test the same flow as mainnet -> Polygon. On the otherhand, USDC on goerli has already been mapped to Mumbai in the old approach so we can still test the bridging of that.
2. Setting up a ENS subdomain: This has already been done in the [ENS proposal](https://github.com/compound-finance/comet/pull/673).

**Transactions**
Goerli proposal execution: https://goerli.etherscan.io/tx/0x5d8a03f2d3a0dc5e34918b1804d58e1e9cbfe63593152f03303ddc05bf5354f0
Mumbai bridged transaction: https://mumbai.polygonscan.com/tx/0x2199393ddd9a17e90bbf748cdca4784c64ba6d29db8c05ea04472b03f8c663e1
Mumbai proposal execution: https://mumbai.polygonscan.com/tx/0x1bbfd1e5396fc78464f829ed17e26d50c0e82c23b32a7546adc93d2a50465921

### CODE REVIEW

File: deployments/mumbai/usdc/migrations/1676659582_configurate_and_ens.ts [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+        args: [ERC20PredicateAddress, USDCAmountToBridge]
+      },
+      // 3. Bridge USDC from mainnet to Mumbai Comet using RootChainManager
+      {
+        target: RootChainManager.address,
+        signature: 'depositFor(address,address,bytes)',
+        calldata: depositForUSDCCalldata
+      },
+      // Note: Cannot test this flow for COMP because Polygon team has changed the way to map new tokens in January 2023
+      // to go through the FxPortal instead of the RootChainManager
```
Comment (jflatow) [inline deployments/mumbai/usdc/migrations/1676659582_configurate_and_ens.ts:133]: 😬

File: deployments/mumbai/usdc/migrations/1676659582_configurate_and_ens.ts [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+    const bridgedUSDC = new Contract(
+      '0x0FA8781a83E46826621b3BC094Ea2A0212e71B23',
+      ERC20__factory.createInterface(),
+      deploymentManager.hre.ethers.provider
+    );
+    expect(await bridgedUSDC.balanceOf(comet.address)).to.be.equal(exp(500, 6));
+
+    // 4. & 5.
+    // const bridgedCOMP = new Contract(
+    //   '0x46DaBF9092B40A3fE6c0bC3331Cd928B600754fE',
```
Comment (jflatow) [inline deployments/mumbai/usdc/migrations/1676659582_configurate_and_ens.ts:218]: What is this address?

File: deployments/mumbai/usdc/migrations/1676659582_configurate_and_ens.ts [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+    expect(await ENSRegistry.ttl(subdomainHash)).to.be.equal(0);
+    expect(officialMarkets).to.deep.equal({
+      5: [
+        {
+          baseSymbol: 'USDC',
+          cometAddress: '0x3EE77595A8459e93C2888b13aDB354017B198188',
+        },
+        {
+          baseSymbol: 'WETH',
+          cometAddress: '0x9A539EEc489AAA03D588212a164d0abdB5F08F5F',
```
Comment (jflatow) [inline deployments/mumbai/usdc/migrations/1676659582_configurate_and_ens.ts:243]: Nice 🏆

File: deployments/mumbai/usdc/migrations/1676659582_configurate_and_ens.ts [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+
+const ENSName = 'compound-community-licenses.eth';
+const ENSResolverAddress = '0x19c2d5D0f035563344dBB7bE5fD09c8dad62b001';
+const ENSRegistryAddress = '0x00000000000C2E074eC69A0dFb2997BA6C7d2e1e';
+const ENSSubdomainLabel = 'v3-additional-grants';
+const ENSSubdomain = `${ENSSubdomainLabel}.${ENSName}`;
+const ENSTextRecordKey = 'v3-official-markets';
+
+const ERC20PredicateAddress = '0xdD6596F2029e6233DEFfaCa316e6A95217d4Dc34';
+const RootChainManagerAddress = '0xBbD7cBFA79faee899Eaf900F13C9065bF03B1A74';
```
Comment (jflatow) [inline deployments/mumbai/usdc/migrations/1676659582_configurate_and_ens.ts:16]: ✅

File: deployments/mumbai/usdc/migrations/1676659582_configurate_and_ens.ts [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+    expect(await daiInfo.supplyCap).to.be.eq(exp(1_000_000, 18));
+    expect(await wbtcInfo.supplyCap).to.be.eq(exp(20_000, 8));
+    expect(await wethInfo.supplyCap).to.be.eq(exp(50_000, 18));
+    expect(await wmaticInfo.supplyCap).to.be.eq(exp(500_000, 18));
+
+    // 2. & 3.
+    // Note: Cannot verify because the USDC we are bridging over is different from the one in Comet
+    // expect(await comet.getReserves()).to.be.equal(exp(400_000, 6));
+    const bridgedUSDC = new Contract(
+      '0x0FA8781a83E46826621b3BC094Ea2A0212e71B23',
```
Comment (jflatow) [inline deployments/mumbai/usdc/migrations/1676659582_configurate_and_ens.ts:210]: ✅

File: deployments/mumbai/usdc/migrations/1676659582_configurate_and_ens.ts (hunk 2 above)
Comment (kevincheng96) [reply deployments/mumbai/usdc/migrations/1676659582_configurate_and_ens.ts:218]: This is the Mumbai COMP that I mapped using the [Polygon mapper](https://mapper.polygon.technology/). Unfortunately, we can't bridge it using the `RootChainManager` approach.

File: deployments/mumbai/usdc/migrations/1676659582_configurate_and_ens.ts [hunk 6]
```diff
@@ -15,7 +15,7 @@ const ENSTextRecordKey = 'v3-official-markets';
 const ERC20PredicateAddress = '0xdD6596F2029e6233DEFfaCa316e6A95217d4Dc34';
 const RootChainManagerAddress = '0xBbD7cBFA79faee899Eaf900F13C9065bF03B1A74';
 
-export default migration('1675200105_configurate_and_ens', {
+export default migration('1676659582_configurate_and_ens', {
```
Comment (jflatow) [inline deployments/mumbai/usdc/migrations/1676659582_configurate_and_ens.ts:18]: 👌

**Comment by kevincheng96** [comment]
Ran the proposal and executed it:

**Transactions**
Goerli proposal execution: https://goerli.etherscan.io/tx/0x5d8a03f2d3a0dc5e34918b1804d58e1e9cbfe63593152f03303ddc05bf5354f0
Mumbai bridged transaction: https://mumbai.polygonscan.com/tx/0x2199393ddd9a17e90bbf748cdca4784c64ba6d29db8c05ea04472b03f8c663e1
Mumbai proposal execution: https://mumbai.polygonscan.com/tx/0x1bbfd1e5396fc78464f829ed17e26d50c0e82c23b32a7546adc93d2a50465921

**Review by jflatow — COMMENTED** [review]
🥇

**Review by jflatow — APPROVED** [review]

**Pull request merged**